components/table/table-wrapper.js

```diff
@@ -359,19 +361,54 @@ export class TableWrapper extends RtlMixin(SelectionMixin(LitElement)) {
 				this._tableIntersectionObserver = new IntersectionObserver((entries) => {
 					entries.forEach((entry) => {
 						if (entry.isIntersecting) {
-							this._applyClassNames(table);
+							this._handleTableChange();
 						}
 					});
 				});
 			} else {
 				this._tableIntersectionObserver.disconnect();
 			}
-			this._tableIntersectionObserver.observe(table);
+			this._tableIntersectionObserver.observe(this._table);
 		}
 
-		this._applyClassNames(table);
+		this._handleTableChange();
+	}
+
+	async _handleTableChange() {
+		await new Promise(resolve => requestAnimationFrame(resolve));
+
+		this._applyClassNames();
+		this._updateStickyTops();
+	}
 
+	_registerMutationObserver(observerName, callback, target, options) {
+		if (this[observerName]) {
+			this[observerName].disconnect();
+		} else if (target) {
+			this[observerName] = new MutationObserver(callback);
+		}
+		if (target) this[observerName].observe(target, options);
 	}
+
+	_updateStickyTops() {
+		if (!this._table || !this.stickyHeaders) return;
+
+		const controlsHeight = this._controls && !this._controls.noSticky ? this._controls.offsetHeight + 4 : 0; // +4 for the shadow's `bottom: -4px`.
+		const borderRadius = this.type === 'default' ? 5 : 0;
+		let rowTop = controlsHeight - borderRadius;
+
+		const stickyRows = Array.from(this._table.querySelectorAll('tr.d2l-table-header, tr[header], thead tr'));
+		stickyRows.forEach(r => {
+			const ths = Array.from(r.querySelectorAll('th'));
+			ths.forEach((th) => {
+				th.style.top = `${rowTop}px`;
```

Comment:
Would expressing this using the existing radius CSS variable work?

```suggestion
				th.style.top = `calc(${rowTop} - var(--d2l-table-border-radius))`;
```

Reply:
Definitely an option - this would mean removing `const borderRadius`. Not sure if there's any performance or readability benefit of doing it this way vs accounting for it in js, what do you think?

Reply:
The big pro to just using `--d2l-table-border-radius` is not needing to duplicate its definition in two places. But yeah, we'd be pushing the calculation from JavaScript down to the browser's CSS engine. I would expect the browser to have clever logic such that it only recalculates when the variable value changes.